### ISSUE TITLE
`jsdoc/no-undefined-types` throws error for local class properties when documenting an exported class

### ISSUE THREAD

**Issue by PartyWumpus** [gajus/eslint-plugin-jsdoc#1771]
## Expected behavior

I should be able to reference a property on the current class I'm documenting without using the class' full name even when it is `export`ed. (as in `{@link property_a}` instead of `{@link MyClassName.property_a}`. (Typescript understands the annotation).

Sorry about all the similar issues.

## Actual behavior

Raises an error `jsdoc/no-undefined-types: The type 'property' is undefined` on the line where the doc is.
(#1762 still happens for exported classes)

## ESLint Config

```js
'jsdoc/no-undefined-types': 'error'
```

## ESLint sample

```typescript
/** xyz {@link property} */
export class MyClass {
	public property: boolean
}
```

## Environment

- Node version: v24.19.0
- ESLint version v10.9.1
- `eslint-plugin-jsdoc` version: 64.3.9

**Comment by brettz9** [comment]
Thanks for the reports!
